contracts/factory/APMRegistryFactory.sol

```diff
@@ -0,0 +1,87 @@
+pragma solidity 0.4.18;
+
+
+import "../apm/APMRegistry.sol";
+import "../ens/ENSSubdomainRegistrar.sol";
+import "../apps/AppProxyFactory.sol";
+
+import "./DAOFactory.sol";
+import "./ENSFactory.sol";
+
+
+contract APMRegistryFactory is DAOFactory, APMRegistryConstants, AppProxyFactory {
+    APMRegistry public registryBase;
+    Repo public repoBase;
+    ENSSubdomainRegistrar public ensSubdomainRegistrarBase;
+    ENS public ens;
+
+    event DeployAPM(bytes32 indexed node, address apm);
+
+    // Needs either one ENS or ENSFactory
+    function APMRegistryFactory(
+        APMRegistry _registryBase,
+        Repo _repoBase,
+        ENSSubdomainRegistrar _ensSubBase,
+        ENS _ens,
+        ENSFactory _ensFactory
+    ) public
+    {
+        registryBase = _registryBase;
+        repoBase = _repoBase;
+        ensSubdomainRegistrarBase = _ensSubBase;
+
+        ens = _ens != address(0) ? _ens : _ensFactory.newENS(this);
```

Comment:
Add a comment to clarify this ternary (I know what it does but it's not super clear immediately)